glue/flagging_script_glue/flagging_e894e5.py

```diff
@@ -1085,20 +1095,14 @@ def name_selector(tokens) -> str:
         'Empty Name' if name is empty.
         id (str): identified last name
     """
-    if tokens == "Empty Name":
-        return tokens
-    # Ex: John Smith Jr
+
+    if tokens == "Empty Name" or tokens == []:
+        return "Empty Name"
+
```

Comment:
This change prevents the same error from occurring.

In order to make sure the index error from returning an empty list doesn't cause the whole job to break, I changed this such that if it happens again, it will just read it as `'Empty Name'`, instead of a null list value.

Reply:
**request:** Can you add more context about what the error is here, so that I (and future parties) can understand exactly what was breaking and what this will fix?

Reply:
Sure. So in `split_logic()`, there are various scenarios where text is split on a certain text match. The error that was happening happened to do with a name like `'A. K. Arnold'`. In the old logic, the `'a k a'` was detected as a match, and split on this name. 

In that part of `split_logic()`, we take the content of whatever was to the _left_  of the split. In the case of  `'A. K. Arnold'`, the left part of the string after the split would have been `null`, or an empty string. Interestingly, this never happened until one problematic name in the condo data came through. So the output `split_logic()` was an empty list, in this case.

Then, the output from `split_logic()` feeds into `name_selector()`, the input was an empty list, and the code from name selector attempts to index a value in the empty list, leading to an index error.

The fix then, just manually adjusts the empty list to an `'Empty Name`' value, which will just lead to a non-matched name.

Let me know if this makes sense @dfsnow

Reply:
Checks out, thanks for the extra context!